pyinfra/modules/virtualenv.py

```diff
@@ -0,0 +1,44 @@
+# pyinfra
+# File: pyinfra/modules/pip.py
+# Desc: manage virtualenvs
+
+'''
+Manage Python virtual environments
+'''
+
+from __future__ import unicode_literals
+
+from pyinfra.api import operation
+from pyinfra.modules import files
+
+
+@operation
+def virtualenv(
+    state, host,
+    path, python='python3', site_packages=False, always_copy=False,
+    present=True,
+):
+    '''
+    Manage virtualenv.
+
+    + python: python interpreter to use
+    + site_packages: give access to the global site-packages
+    + always_copy: always copy files rather than symlinking
+    + present: whether the virtualenv should be installed
+    '''
+
+    if present is False and host.fact.directory(path):
+        # Ensure deletion of unwanted virtualenv
+        # no 'yield from' in python 2.7
+        for cmd in files.directory(state, host, path, present=False):
```

Comment:
I came across this when switching to the `yield` everywhere! You can just do `yield files.directory...` as pyinfra will "unroll" the generator itself @ https://github.com/Fizzadar/pyinfra/blob/develop/pyinfra/api/util.py#L25.